ios/brave-ios/Sources/Brave/Frontend/Sync/BraveCore/History/HistoryAPIExtensions.swift

```diff
@@ -67,4 +67,22 @@ extension BraveHistoryAPI {
       }
     }
   }
+
+  @MainActor func search(
+    withQuery query: String?,
+    options: HistorySearchOptions
+  ) async -> [HistoryNode] {
+    await withCheckedContinuation { continuation in
```

Comment:
I realize this is copy pasted, but for the second usage you've moved cancellation lifetime from the HistoryModel to the call itself, so it can never cancel until its returned. If we do want to introduce an async variant of this method we should introduce proper cancellation through `withTaskCancellationHandler`:

```swift
var cancellable: HistoryCancellable?
return await withTaskCancellationHandler {
  cancellable?.cancel()
} operation: {
  try Task.checkCancellation()
  return withCheckedContinuation {
    cancellable = ...
  }
}
```